src/Enterspeed.Source.UmbracoCms/Handlers/PreviewMedia/EnterspeedPreviewMediaTrashedJobHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Net;
+using Enterspeed.Source.Sdk.Api.Services;
+using Enterspeed.Source.UmbracoCms.Data.Models;
+using Enterspeed.Source.UmbracoCms.Models;
+using Enterspeed.Source.UmbracoCms.Exceptions;
+using Enterspeed.Source.UmbracoCms.Providers;
+using Umbraco.Cms.Core.Services;
+
+namespace Enterspeed.Source.UmbracoCms.Handlers.PreviewMedia
+{
+    public class EnterspeedPreviewMediaTrashedJobHandler : IEnterspeedJobHandler
+    {
+        private readonly IEnterspeedIngestService _enterspeedIngestService;
+        private readonly IEnterspeedConnectionProvider _enterspeedConnectionProvider;
+        private readonly IMediaService _mediaService;
+
+        public EnterspeedPreviewMediaTrashedJobHandler
+        (
+            IEnterspeedIngestService enterspeedIngestService,
+            IEnterspeedConnectionProvider enterspeedConnectionProvider,
+            IMediaService mediaService)
+        {
+            _enterspeedIngestService = enterspeedIngestService;
+            _enterspeedConnectionProvider = enterspeedConnectionProvider;
+            _mediaService = mediaService;
+        }
+
+        public bool CanHandle(EnterspeedJob job)
+        {
+            return
+                _enterspeedConnectionProvider.GetConnection(ConnectionType.Preview) != null
+                && job.EntityType == EnterspeedJobEntityType.Media
+                && job.JobType == EnterspeedJobType.Delete
+                && job.ContentState == EnterspeedContentState.Preview;
+        }
+
+        public void Handle(EnterspeedJob job)
+        {
+            var parsed = int.TryParse(job.EntityId, out var parsedId);
+            var media = parsed ? _mediaService.GetById(parsedId) : null;
```

Comment:
Why do we load the umbraco media. We only use the media.id but we already have the id from the job?

Also _enterspeedIngestService.Delete(media?.Id.ToString(), .....) could result in us sending null as the id parameter how will the enterspeedIngestService.Delete method handle that?

Reply:
You are right. This seems odd. This is something the has been living for a while in the default way we handled media (Not only for preview). We should probably change that for both the default way we handle media, as well as the preview functionality. What do you think?

Reply:
I have fixed it for all job handlers that handles delete now :)